care/emr/utils/reset_password.py

```
@@ -0,0 +1,100 @@
+            return (None, "Token invalid due to password change")
+
+        if payload.get("type") != "password_reset":
+            return (None, "Invalid token type")
+        if payload.get("sub") != str(user.external_id):
+            return (None, "Token doesn't match user")
+
+    except jwt.ExpiredSignatureError:
+        error_message = "Token has expired"
+    except User.DoesNotExist:
+        error_message = "User not found"
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)